sdk/eventhub/azure-eventhub/azure/eventhub/_transport/_uamqp_transport.py

```diff
@@ -384,8 +384,7 @@ def send_messages(producer, timeout_time, last_exception, logger):
                     raise producer._condition
 
         @staticmethod
-        def set_message_partition_key(message, partition_key, **kwargs):  # pylint:disable=unused-argument
-            # type: (Message, Optional[Union[bytes, str]], Any) -> Message
+        def set_message_partition_key(message: Message, partition_key: Union[bytes, str], **kwargs: Any) -> Message:  # pylint:disable=unused-argument
```

Comment:
should `partition_key` be Optional like it was previously? It looks like we do an optional check in the code